nix/pkgs/gomobile/default.nix

```
@@ -32,6 +32,21 @@ buildGo122Module {
 
   CGO_ENABLED = "1";
 
+  preBuild = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    echo "Configuring environment for gomobile iOS builds..."
+
+    # to fix : error: unable to find sdk: 'iphoneos'
+    export DEVELOPER_DIR="/Applications/Xcode.app/Contents/Developer";
+    export SDKROOT="$(/usr/bin/xcrun --sdk macosx --show-sdk-path)";
+
+    if xcrun --sdk iphoneos --show-sdk-path >/dev/null 2>&1; then
+      echo "iOS SDK found at: $(xcrun --sdk iphoneos --show-sdk-path)"
+    else
+      echo "iOS SDK not found. iOS builds will fail."
+      echo "Please ensure Xcode is installed and run: sudo xcode-select --switch /Applications/Xcode.app"
+    fi
+  '';
```

Way too much noise, less is more:
```suggestion
  preBuild = lib.optionalString stdenv.hostPlatform.isDarwin ''
    # to fix : error: unable to find sdk: 'iphoneos'
    export DEVELOPER_DIR="/Applications/Xcode.app/Contents/Developer";
    export SDKROOT="$(/usr/bin/xcrun --sdk macosx --show-sdk-path)";
    if ! xcrun --sdk iphoneos --show-sdk-path >/dev/null 2>&1; then
      echo "iOS SDK not found. Ensure Xcode is installed: sudo xcode-select --switch /Applications/Xcode.app"
    fi
  '';
```
Also, isn't installing iOS SDK a separate step too?

---

iOS SDK is part of xcode setup.
installing an iOS Runtime is a manual setup, which we do in `infra-ci`

---

Fair enough.